web/tests/hashchange.test.js

```diff
@@ -78,6 +79,29 @@ run_test("terms_round_trip", () => {
     assert.deepEqual(narrow, [{operator: "stream", operand: "Florida, USA", negated: false}]);
 });
 
+run_test("stream_to_channel_rename", () => {
```

Comment:
Added this test to make it clear in this interim state what's happening with these functions as they interact together.

Reply:
Yeah, this is very helpful.